orangespaceman/phx
# Update Python dependencies

I've updated the python dependencies and done a bit of testing (more details below). Do you think there's anything else in particular worth testing before merging this?

## Update

I update all the backend dependencies by doing:

```
cd requirements
rm *.txt
make all
```

and

```
pre-commit autoupdate
```

## Django

This includes a major version bump of Django to [5.0](https://docs.djangoproject.com/en/5.0/releases/5.0/)

Everything seems to run fine. The only difference I've spotted is that the admin interface looks a little different and now has a dark mode :night_with_stars:

![image](https://github.com/orangespaceman/phx/assets/3168140/ce917c1c-9bea-4426-a32c-6fa5ada2b3ef)

![Screenshot from 2024-06-06 20-07-34](https://github.com/orangespaceman/phx/assets/3168140/db780e4d-e5ed-473e-91b3-b0943cded8d7)

Might be worth browsing through the [backwards incompatible changes](https://docs.djangoproject.com/en/5.0/releases/5.0/#backwards-incompatible-changes-in-5-0) in-case anything jumps out to you

## Testing

- Download the dump you shared from a few weeks ago
- Ran the site locally
- Navigated through various pages
- Logged in as admin
- Added a new result

All seemed to work smoothly

![Screenshot from 2024-06-06 20-14-12](https://github.com/orangespaceman/phx/assets/3168140/9defe55d-3961-4531-8149-8852d9e38c6d)

Upon starting the server there is a new warning:

```
?: (ckeditor.W001) django-ckeditor bundles CKEditor 4.22.1 which isn't supported anmyore and which does have unfixed security issues, see for example https://ckeditor.com/cke4/release/CKEditor-4.24.0-LTS . You should consider strongly switching to a different editor (maybe CKEditor 5 respectively django-ckeditor-5 after checking whether the CKEditor 5 license terms work for you) or switch to the non-free CKEditor 4 LTS package. See https://ckeditor.com/ckeditor-4-support/ for more on this. (Note! This notice has been added by the django-ckeditor developers and we are not affiliated with CKSource and were not involved in the licensing change, so please refrain from complaining to us. Thanks.)
```

It looks like CKEditor 5 is [not a simple drop-in replacement](https://ckeditor.com/docs/ckeditor5/latest/updating/ckeditor4/migration-from-ckeditor-4.html) so I think it's worth doing separately to the rest of these updates.


Ran the tests locally, they passed but with a quite a few warnings that needed to be fixed. Mostly this one:

```
DeprecationWarning: FixtureFactory._after_postgeneration will stop saving the instance after postgeneration hooks in the next major release.
  If the save call is extraneous, set skip_postgeneration_save=True in the FixtureFactory.Meta.
  To keep saving the instance, move the save call to your postgeneration hooks or override _after_postgeneration.
    warnings.warn(
```



I've taken the [recommended action ](https://factoryboy.readthedocs.io/en/latest/changelog.html) and `set skip_postgeneration_save=True`. I didn't quite understand what was meant though so you might want to double check

There are two remaining that are coming from dependencies so still looking at how best to fix them:

```
../env/lib/python3.10/site-packages/monkeybiz/__init__.py:3
  /home/lachie/Code/phx2/env/lib/python3.10/site-packages/monkeybiz/__init__.py:3: DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html
    import pkg_resources

phx/gallery/tests/views/test_gallery_detail_view.py::TestGalleryDetailView::test_gallery_images
  /home/lachie/Code/phx2/env/lib/python3.10/site-packages/easy_thumbnails/storage.py:27: RemovedInDjango51Warning: django.core.files.storage.get_storage_class is deprecated in favor of using django.core.files.storage.storages.
    self._wrapped = get_storage_class(

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
```

There's an open easy_thumbnails PR which might fix that one

There's also an open Issue about the monkeybiz one but that project doesn't look very active. It's a dependency of `django-nested-admin` so maybe they will remove it.

For now I've suppressed them with these two commits:

https://github.com/orangespaceman/phx/pull/52/commits/5b3af31cc3ca53a0d4d5742526a8c22f420fab64
https://github.com/orangespaceman/phx/pull/52/commits/4f11c4c8fca14eb3e5ef385fd4b94152320917bf
